pkg/runtime/runtime.go

```
@@ -104,11 +104,24 @@ func Register(ext string, r Interface) {
 	runtimes[ext] = r
 }
 
-// Lookup returns a runtime by path.
-func Lookup(path string) (Interface, bool) {
-	ext := filepath.Ext(path)
-	r, ok := runtimes[ext]
-	return r, ok
+// Lookup returns a runtime by kind and path.
+// If an extension match is found, use that runtime. Otherwise rely on the task kind.
+func Lookup(kind api.TaskKind, path string) (Interface, bool) {
+	pathExt := filepath.Ext(path)
+	possible := []Interface{}
+	for ext, runtime := range runtimes {
+		if runtime.Kind() != kind {
+			continue
+		}
+		if pathExt == ext {
+			return runtime, true
+		}
+		possible = append(possible, runtime)
+	}
+	if len(possible) != 1 {
+		return nil, false
```

Should we return an error here instead?

---

Hrm yeah I'll just make the func return error instead of bool